tests/syslog/test_syslog_source_ip.py

```diff
@@ -0,0 +1,560 @@
+
+MGMT_IP_ADDRESSES = {"ipv4": {"syslog_server_ip": "", "syslog_server_mac": "90:90:90:90:90:31",
+                              "source_ip": ""},
+                     "ipv6": {"syslog_server_ip": "", "syslog_server_mac": "90:90:90:90:90:32",
+                              "source_ip": ""}}
+
+VRF_LIST = ["default", "Vrf-data", "mgmt"]
+
+SYSLOG_TEST_DATA = {"default": DEFAULT_VRF_IP_ADDRESSES,
+                    "Vrf-data": DATA_VRF_IP_ADDRESSES,
+                    "mgmt": MGMT_IP_ADDRESSES}
+
+SYSLOG_CONFIG_COMBINATION = {
+    "vrf_unset_source_unset_port_None": {"is_set_vrf": False, "is_set_source": False, "port": None,
+                                         "vrf_list": VRF_LIST[0:1]},
+    "vrf_unset_source_unset_port_600": {"is_set_vrf": False, "is_set_source": False, "port": 600,
+                                        "vrf_list": VRF_LIST[0:1]},
+    "vrf_unset_source_set_port_None": {"is_set_vrf": False, "is_set_source": True, "port": None,
+                                       "vrf_list": VRF_LIST[0:1]},
+    "vrf_unset_source_set_port_650": {"is_set_vrf": False, "is_set_source": True, "port": 650,
+                                      "vrf_list": VRF_LIST[0:1]},
+    "vrf_set_source_unset_None": {"is_set_vrf": True, "is_set_source": False, "port": None, "vrf_list": VRF_LIST},
+    "vrf_set_source_unset_700": {"is_set_vrf": True, "is_set_source": False, "port": 700, "vrf_list": VRF_LIST},
+    "vrf_set_source_set_None": {"is_set_vrf": True, "is_set_source": True, "port": None, "vrf_list": VRF_LIST},
+    "vrf_set_source_set_800": {"is_set_vrf": True, "is_set_source": True, "port": 800, "vrf_list": VRF_LIST}}
+
+SYSLOG_CONFIG_COMBINATION_CASE = ["vrf_unset_source_unset_port_None",
+                                  "vrf_unset_source_unset_port_600",
+                                  "vrf_unset_source_set_port_None",
+                                  "vrf_unset_source_set_port_650",
+                                  "vrf_set_source_unset_None",
+                                  "vrf_set_source_unset_700",
+                                  "vrf_set_source_set_None",
+                                  "vrf_set_source_set_800"]
+
+SYSLOG_DEFAULT_PORT = 514
+
+
+@pytest.fixture(scope="module", autouse=True)
+def is_support_ssip(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
+    """
+    Check if the image support ssip feature. If no, skip all tests
+
+    Args:
+        duthosts: DUT hosts fixture
+        enum_rand_one_per_hwsku_frontend_hostname: DUT fixture
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    show_syslog_res = duthost.command('show syslog', module_ignore_errors=True)['stderr_lines']
+    if show_syslog_res:
+        pytest.skip("This image doesn't support ssip feature, so skipp all related tests")
+
+
+@pytest.fixture(scope="module", autouse=True)
+def restore_config_by_config_reload(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
```

Comment:
@JibinBao can we move all auxiliary stuff (fixtures, helpers, etc.) out of this file? IMHO, it will simply code understanding a lot

Reply:
Maybe we can move it to the conftest, because there is another existing script that will not use the fixture.

Reply:
I want to keep it here